File: GrooveOn.Models/ResponseObjects/ArtistResponse.cs
using System;

namespace GrooveOn.Model.ResponseObject
{
    public class ArtistResponse
    {
        public int Id { get; set; }

        public string? ExternalArtistId { get; set; }

        public string Source { get; set; } = string.Empty;

        public string Name { get; set; } = string.Empty;

        public string? Picture { get; set; }

        public DateTime CreatedAt { get; set; }
    }
}

File: UI/grooveon_desktop/lib/models/response/artist_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'artist_response.g.dart';

@JsonSerializable()
class ArtistResponse {
  final int id;
  final String? externalArtistId;
  final String source;
  final String name;
  final String? picture;
  final DateTime createdAt;

  ArtistResponse({
    required this.id,
    this.externalArtistId,
    required this.source,
    required this.name,
    this.picture,
    required this.createdAt,
  });

  factory ArtistResponse.fromJson(Map<String, dynamic> json) =>
      _$ArtistResponseFromJson(json);

  Map<String, dynamic> toJson() => _$ArtistResponseToJson(this);
}
